dds/DCPS/security/CommonUtilities.cpp

```diff
@@ -68,6 +68,22 @@ bool set_security_error(DDS::Security::SecurityException& ex,
   return false;
 }
 
+bool set_security_error(DDS::Security::SecurityException& ex,
+                        int code,
+                        int minor_code,
+                        const char* message_prefix,
+                        unsigned long openssl_error)
+{
+  using DCPS::String;
+  using DCPS::to_dds_string;
+  ex.code = code;
+  ex.minor_code = minor_code;
+  // OpenSSL errors are rendered in hex, matching "openssl errstr"
+  static const bool as_hex = true;
+  ex.message = (message_prefix + String(" OpenSSL error ") + to_dds_string(openssl_error, as_hex)).c_str();
```

Comment:
Why not covert the error to a descriptive string?

Reply:
Could be done in a separate PR if that's needed.  Since we're currently logging the error number I went with that.  It's now in one place so not hard to change